sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -393,36 +420,268 @@ def tune(
+                    if old_attr is not None:
+                        value = f"${{trialParameters.{p_name}}}"
+                    setattr(lora_config, p_name, value)
+                    p_value.name = p_name
+                    experiment_params.append(p_value)
+                    trial_params.append(
+                        models.V1beta1TrialParameterSpec(name=p_name, reference=p_name)
+                    )
+                elif p_value is not None:
+                    old_attr = getattr(lora_config, p_name, None)
+                    if old_attr is not None:
+                        value = type(old_attr)(p_value)
+                    setattr(lora_config, p_name, value)
+
+            init_container_spec = utils.get_container_spec(
+                name=STORAGE_INITIALIZER,
+                base_image=STORAGE_INITIALIZER_IMAGE,
+                args=[
+                    "--model_provider",
+                    mp,
+                    "--model_provider_parameters",
+                    json.dumps(
+                        model_provider_parameters.__dict__, cls=utils.SetEncoder
+                    ),
+                    "--dataset_provider",
+                    dp,
+                    "--dataset_provider_parameters",
+                    json.dumps(dataset_provider_parameters.__dict__),
+                ],
+                volume_mounts=[STORAGE_INITIALIZER_VOLUME_MOUNT],
+            )
+
+            lora_config = json.dumps(lora_config.__dict__, cls=utils.SetEncoder)
+            training_args = json.dumps(training_args.to_dict())
+
+            container_spec = utils.get_container_spec(
+                name=constants.DEFAULT_PRIMARY_CONTAINER_NAME,
+                base_image=TRAINER_TRANSFORMER_IMAGE,
+                args=[
+                    "--model_uri",
+                    model_provider_parameters.model_uri,
+                    "--transformer_type",
+                    model_provider_parameters.transformer_type.__name__,
+                    "--model_dir",
+                    VOLUME_PATH_MODEL,
+                    "--dataset_dir",
+                    VOLUME_PATH_DATASET,
+                    "--lora_config",
+                    f"'{lora_config}'",
+                    "--training_parameters",
+                    f"'{training_args}'",
+                ],
+                volume_mounts=[STORAGE_INITIALIZER_VOLUME_MOUNT],
+                resources=resources_per_trial,
+            )
+
+            storage_initializer_volume = models.V1Volume(
+                name=STORAGE_INITIALIZER,
+                persistent_volume_claim=models.V1PersistentVolumeClaimVolumeSource(
+                    claim_name=name
+                ),
+            )
+
+            pod_spec = utils.get_pod_template_spec(
+                containers=[container_spec],
+                init_containers=[init_container_spec],
+                volumes=[storage_initializer_volume],
+                restart_policy="Never",
+            )
+
```

Comment:
We are still creating Job instead of PyTorchJob in the Trial template.
Did you push your latest changes @helenxie-bit ?

Reply:
@andreyvelich I’ve pushed my changes in a separate PR: https://github.com/kubeflow/katib/pull/2409. I’m not sure I fully understood your suggestions, so could you please refer to this comment: https://github.com/kubeflow/katib/pull/2393#discussion_r1724681935?